trunk/SUAVE/Plots/Geometry_Plots/plot_vehicle.py

```diff
@@ -39,286 +41,470 @@ def plot_vehicle(vehicle, save_figure = False, plot_control_points = True, save_
+    """ This plots a 3D surface of the  propeller
+
+    Assumptions: 
+    None
+
+    Source:   
+    None
+    
+    Inputs:   
+    VD                   - vortex distribution    
+    propulsor            - propulsor data structure 
+    
+    Properties Used:
+    N/A
+    """          
+        
+    # unpack         
+    num_B  = prop.number_of_blades      
+    a_sec  = prop.airfoil_geometry          
+    a_secl = prop.airfoil_polar_stations
+    beta   = prop.twist_distribution         
+    b      = prop.chord_distribution         
+    r      = prop.radius_distribution 
+    MCA    = prop.mid_chord_aligment
+    t      = prop.max_thickness_distribution
+    ta     = -propulsor.thrust_angle
+    
+    if isinstance(propulsor,Lift_Cruise):
+        if propulsor_name == 'propeller': 
+            origin = propulsor.propeller.origin
             
-            # create empty arrays for storing geometry
-            G = Data()
-            G.XA1 = np.zeros((dim-1,af_pts))
-            G.YA1 = np.zeros_like(G.XA1)
-            G.ZA1 = np.zeros_like(G.XA1)
-            G.XA2 = np.zeros_like(G.XA1)
-            G.YA2 = np.zeros_like(G.XA1)
-            G.ZA2 = np.zeros_like(G.XA1)
-            G.XB1 = np.zeros_like(G.XA1)
-            G.YB1 = np.zeros_like(G.XA1)
-            G.ZB1 = np.zeros_like(G.XA1)
-            G.XB2 = np.zeros_like(G.XA1)
-            G.YB2 = np.zeros_like(G.XA1)
-            G.ZB2 = np.zeros_like(G.XA1)  
+        elif propulsor_name == 'rotor': 
+            origin = propulsor.rotor.origin
+    else:
+        origin = prop.origin
+    n_points  = 10
+    af_pts    = (2*n_points)-1
+    dim       = len(b)
+    num_props = len(origin) 
+    theta     = np.linspace(0,2*np.pi,num_B+1)[:-1]   
+    
+    # create empty arrays for storing geometry
+    G = Data()
+    G.XA1 = np.zeros((dim-1,af_pts))
+    G.YA1 = np.zeros_like(G.XA1)
+    G.ZA1 = np.zeros_like(G.XA1)
+    G.XA2 = np.zeros_like(G.XA1)
+    G.YA2 = np.zeros_like(G.XA1)
+    G.ZA2 = np.zeros_like(G.XA1)
+    G.XB1 = np.zeros_like(G.XA1)
+    G.YB1 = np.zeros_like(G.XA1)
+    G.ZB1 = np.zeros_like(G.XA1)
+    G.XB2 = np.zeros_like(G.XA1)
+    G.YB2 = np.zeros_like(G.XA1)
+    G.ZB2 = np.zeros_like(G.XA1)  
+    
+    for n_p in range(num_props):  
+        rot    = prop.rotation[n_p] 
+        a_o    = 0
+        flip_1 = (np.pi/2)  
+        flip_2 = (np.pi/2) 
         
-            for n_p in range(num_props):  
-                rot    = prop.rotation[n_p] 
-                a_o    = 0
-                flip_1 = (np.pi/2)  
-                flip_2 = (np.pi/2) 
+        for i in range(num_B):   
+            # get airfoil coordinate geometry     
+            airfoil_data = import_airfoil_geometry(a_sec,npoints=n_points)   
+            
+            # store points of airfoil in similar format as Vortex Points (i.e. in vertices)
+            for j in range(dim-1): # loop through each radial station 
```

Comment:
this loop is vectorizable.

Reply:
lines 393 and 394 prevent this. 
```
ixpts = airfoil_data.x_coordinates[a_secl[j]]
izpts = airfoil_data.y_coordinates[a_secl[j]]
```

Even if you create a vector of geometry points at the beginning, one still needs to index a particular airfoil.

Reply:
Here: np.take(airfoil_data.x_coordinates,a_secl,axis=0)

Reply:
Thanks, no more inner loop. Issue resolved